neonvm/controllers/metrics.go

```diff
@@ -156,36 +162,48 @@ func (d *wrappedReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	defer d.lock.Unlock()
 	outcome := SuccessOutcome
 	if err != nil {
-		d.failing[req.NamespacedName] = struct{}{}
 		if errors.IsConflict(err) {
 			outcome = ConflictOutcome
+			d.conflicting[req.NamespacedName] = struct{}{}
```

Comment:
need to additionally `delete(...)` failing/conflicting when there's an error in the other one, right?

Reply:
I don't think so, if it keeps failing, and a conflict occurs, we want it to keep being registered as failing.

Reply:
Hm, I see. Fair enough. Does it make sense to go the other way as well (i.e. conflict -> error continues registering as conflict)

Reply:
Ok, let's imagine the VM keeps conflicting, and there are no other failures -> it will register as conflicting. 
But, if the failure were to arise (e.g. QMP errors), then it will stop being conflicting, and starts to be failing only.
This means that an unrelated error impacts the semantics of "conflict", which is why I think it should be kept as-is.

Also, it is simpler to reason, and we might at some point decide to handle have more error types, and it is easier to have the semantics that different modes of failure are independent, and can only be finished when a successful reconciliation occurs.

Reply:
My mental model for conflicts vs errors is that conflicts are practically guaranteed to be transient (although we should still track in practice because perhaps there's a bug), whereas errors _may_ be persistent.

So if a VM has several conflicts, but then starts getting errors, I would assume it's more likely that the conflicts were resolved than the errors being spurious (i.e. IMO it would make more sense to continue tracking as a conflict when there's errors)